_posts/2020-12-24-how-to-start-with-foreman

```diff
@@ -0,0 +1,399 @@
+---
+layout: post
+title: How to start with Foreman
+date: 2020-12-24 17:00:00
+author: Marek Hulan
+tags:
+- foreman
+- community
+- tutorial
+- howto
+- beginners
+---
+
+This post targets primarily users new to Foreman. It can be interesting for those,
+who would like to start using Foreman but feel lost or who just installed Foreman
+and would like to investigate its capabilities. We'll assume a fresh Foreman
+installation without any plugin and with the default configuration. The primary focus 
+will be on on filling in the data about the infrastructure. If you're searching for
+installation instructions, please take a look at [this chapter](https://theforeman.org/manuals/2.3/index.html#3.InstallingForeman)
+of the Foreman manual. This post assumes installation using the foreman-installer.
+
+<!--more-->
+
+Foreman allows to define many objects, which serve different purposes. But the
+heart of everything is the Host object, It glues all other objects together.
+Host represents some machine. It can be a device on network or simply something,
+that you want to install operating system. It's typically a thing you want to manage
+or automate some tasks around it.
+In virtualization terminology, it can be host but also a guest. It can be a baremetal
+machine, it can be VM running on some hypervisor or a VM somewhere in the cloud.
+While Foreman does not primarily target these, it can also be a router, IoT sensor,
+laptop, container.
+
+Regardless of what kind of machine the host is, Foreman calls that simply the Host. Hosts have name.
+They also must belong to some organization and location. This is the bare minimum
+that Foreman requires in order to define a Host. Some use-cases may require more
+attributes. E.g. to provision a host in the cloud, cloud credentials are required
+or in order to provision a host using PXE boot, host MAC address is needed. In order
+to run job though SSH on it, it's FQDN or IP must be set.
+
+Hosts can be either *managed* or *unmanaged*. While it's a simple boolean flag, it
+dramatically changes how Foreman treats them. Managed hosts are seen
+as something, that Foreman tries to manage in external services. Imagine you
+create a new Host in Foreman, you set it's name (FQDN), IP address, MAC address.
+Based on the managed flag, Foreman can reach out to associated DHCP server to create
+a reservation for the given MAC and IP pair. It would try to create A, AAAA and PTR records
+for the hostname and IP address pair. What exactly happens is driven by what is configured
+in Foreman. We won't go to more details in this post,
+but reading our [manual](https://theforeman.org/manuals/latest/#4.GeneralForeman)
```

Comment:
you link to latest here, but 2.3 above, I don't mind either, but it should be consistent ;)

Reply:
fixed above, I'd use latest everywhere, but only foreman has that symlink